app/src/backend/environment/hooks/useEnvironmentManager.ts

```diff
@@ -38,9 +39,10 @@ const VARIABLES_PRECEDENCE_ORDER = ["ENVIRONMENT", "COLLECTION"];
 
 const useEnvironmentManager = () => {
   const dispatch = useDispatch();
+  const apiClientRecords = useSelector(getAllRecords);
   const [isLoading, setIsLoading] = useState(false);
   const [isEnvironmentsDataLoaded, setIsEnvironmentsDataLoaded] = useState(false);
-  const { apiClientRecords, onSaveRecord } = useApiClientContext();
+  const { onSaveRecord } = useApiClientContext();
```

Comment:
Lets not change the DX for this, get the `apiClientRecords` from redux and return from `useApiClientContext`

```suggestion
const { apiClientRecords, onSaveRecord } = useApiClientContext();